c++.html.markdown

```diff
@@ -249,6 +250,56 @@ const string& barRef = bar; // Create a const reference to bar.
 // Like C, const values (and pointers and references) cannot be modified.
 barRef += ". Hi!"; // Error, const references cannot be modified.
 
+// Sidetrack: Before we talk more about references, we must introduce a concept
+// called a temporary object. Suppose we have the following code:
+string tempObjectFun() { ... }
+string retVal = tempObjectFun();
+
+// What happens in the second line is actually:
+//   - a string object is returned from `tempObjectFun`
+//   - a new string is constructed with the returned object as arugment to the
+//     constructor
+//   - the returned object is destroyed
```

Comment:
https://github.com/adambard/learnxinyminutes-docs/commit/85f6ba0b57b9d894c694df66449b1e1c555c625b

@vendethiel How's this? I accidentally pushed directly, didn't make a PR. Feel free to comment on the commit itself.